Add Fraud Detection example (#148)
vaticle/typedb-examples: grc.fraudDetection_web/README.md

# fraudDetection_web

This project uses Quarkus, the Supersonic Subatomic Java Framework in order to let us present results with GraphQL.
In this project, we use TypeDB to represent a queryable database of relevant fraud detection data about some transactions between Cardholders and Companies.

## Introduction

We have an imaginary transaction dataset and our application implements some very basic functions:

- Search for Bank
- Search for Cardholder
- Search for Company
- Search for Transaction 

We are demonstrating sub-typing, powerful rules and rule combination in our schema. 
####

> We are also how easy it is to create complex queries using query composition.

Reviewer: ```suggestion
We can also see how easy it is to create complex queries using query composition.
```

####

> We are using the DAO design pattern, we have following components on which our design depends:

Reviewer: ```suggestion
We are using the DAO design pattern, we have the following components on which our design depends:
```

- The model which is transferred from one layer to the other.

> - The interfaces which provides a flexible design.

Reviewer: ```suggestion
- The interfaces which provide a flexible design.
```

- The interface implementation which is a concrete implementation of the persistence logic.

> The querry composition can be observed in all DAOs.

Reviewer: ```suggestion
The query composition can be observed in all DAOs.
```

> ```javascript

Reviewer: ```suggestion
```java
```

String getQueryStr = "match " + TX_MATCH + CardholderDao.CARDHOLDER_MATCH + BankDao.BANK_MATCH + MerchantDao.MERCHANT_MATCH;
```

## Running the application in dev mode

> 1. Checkout this repository: `git clone https://github.com/vaticle/typedb-examples && cd typedb-examples`.

Reviewer: ```suggestion
1. Checkout this repository: `git clone https://github.com/vaticle/typedb-examples && cd typedb-examples/grc.fraudDetection_web`.
```

Reviewer: This also raises the fact we should have a better name and structure for where this example fits into the rest of typedb-examples.

This could fit under our existing commerce package? But perhaps a `security` or `cybersecurity` module would be more appropriate.

2. Start the [TypeDB Server](http://docs.vaticle.com/docs/running-typedb/install-and-run#start-the-typedb-server). Check that it's listening to address: `0.0.0.0:1729`.
3. You can run your application in dev mode that enables live coding using:
```shell script
./mvnw compile quarkus:dev
```
> **_NOTE:_**  Quarkus now ships with a Dev UI, which is available in dev mode only
> at http://localhost:8080/q/dev/.

4. You can then choose the GraphQL UI.
5. You can now use this interface to query the database (you can find all available queries in the top right corner).

## TypeDB Description

### Schema

> The schema is stored in the `schema.tql` file.

Reviewer: ```suggestion
The schema is stored in the `schema.tql` file under src/main/resources/.
```

Reviewer: Or even better, provide a link to it. You can do this using a relative hyperlink, e.g.:
```
[something](src/main/something.tql)
```
